scripts/add_electricity.py

```diff
@@ -385,6 +385,38 @@ def attach_wind_and_solar(
                 efficiency=costs.at[suptech, "efficiency"],
             )
 
+            if tech == "csp":  # add store and link for CSP
+                csp_buses_i = n.madd(
+                    "Bus",
+                    ds.indexes["bus"] + " " + tech,
+                    carrier=tech,
+                    unit="MW",
+                )
+
+                n.madd(
+                    "Store",
+                    csp_buses_i,
```

Comment:
The workflow expects to have here an index-like object, while `csp_bises_i` have been defined earlier as `PyPSA.Network` object which can lead to issues.

Reply:
Thanks for the clarity. I used indexes but switched them at some point. I would change It back.

Reply:
I have removed the conditional and refined the implementation here.